packages/svelte/src/compiler/phases/3-transform/client/visitors/EachBlock.js

```diff
@@ -186,8 +186,23 @@ export function EachBlock(node, context) {
 	if (invalidate_store) sequence.push(invalidate_store);
 
 	if (node.context.type === 'Identifier') {
+		const binding = /** @type {Binding} */ (context.state.scope.get(node.context.name));
+
 		child_state.transform[node.context.name] = {
-			read: (flags & EACH_ITEM_REACTIVE) !== 0 ? get_value : (node) => node,
+			read: (node) => {
+				if (binding.reassigned) {
+					// we need to do `array[$$index]` instead of `$$item` or whatever
+					// TODO this only applies in legacy mode, reassignments are
```

Comment:
what about this todo?

Reply:
it's a reminder to remove it in 6.0 or whenever we remove legacy support. i'll make that explicit

```suggestion
					// TODO 6.0 this only applies in legacy mode, reassignments are
```